server/src/main/java/org/eclipse/hono/server/MessageForwardingEndpoint.java

```diff
@@ -38,13 +40,24 @@
  */
 public abstract class MessageForwardingEndpoint extends BaseEndpoint {
 
+    protected CounterService              counterService;
```

Comment:
simply by means of something like

````
public abstract class MessageForwardingEndpoint extends BaseEndpoint {
 
    private CounterService              counterService = EmptyCounterService.instance();
...
}
````

The setter could still be used to overwrite the default implementation. We would also provide a getter in order to prevent subclasses from setting the field to `null`.